File: src/math.go
// check if a number is a prime

func IsPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	for i := 3; i*i <= n; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// IsPrime2 is a more efficient version of IsPrime
func IsPrime2(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	if n < 9 {
		return true
	}
	if n%3 == 0 {
		return false
	}
	r := int(n-1) / 6
	f := 5
	for f*f <= n {
		if n%f == 0 {
			return false
		}
		if n%(f+2) == 0 {
			return false
		}
		f += 6
	}
	return true
}

File: src/math.js
// return if a number is odd or even
export function isOdd(num) {
    return num % 2 == 1;
}

// check if a number is prime
export function isPrime(num) {
    if (num == 1) return false;
    for (let i = 2; i < num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}

// return the factorial of a number
export function factorial(num) {
    if (num == 1) return 1;
    return num * factorial(num - 1);
}
